airbyte-cdk/java/airbyte-cdk/core/src/test/kotlin/io/airbyte/cdk/integrations/destination/async/AsyncStreamConsumerTest.kt

```diff
@@ -732,4 +733,38 @@ class AsyncStreamConsumerTest {
             }
         assertEquals(expRecords, actualRecords)
     }
+
+    @Test
+    internal fun deserializeAirbyteMessageWithUnrecognizedType() {
+        val airbyteMessage = AirbyteMessage().withType(AirbyteMessage.Type.RECORD)
+        val serialized = Jsons.serialize(airbyteMessage)
```

Comment:
why not just directly hardcode the json string? I feel like going through the message -> serialize -> replace makes this test less clear
```kotlin
val retyped = """{"type": "__UNKNOWN_TYPE_OF_MESSAGE__"}"""
```

Reply:
Technically I don't know if the builder is adding any other metadata to make it a legal message?

Reply:
I think either way is fine. The error and catching is explicit enough that only the relevant issues get filtered out